File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] arr=new int[2];
        arr[0]=BS(nums,target,true);
        arr[1]=-1;
        if(arr[0]!=-1){
            arr[1]=BS(nums,target,false);
        }return arr;
        
        
    }
    
    int BS(int[] nums,int target,boolean start){
        int st=0;
        int en=nums.length-1,ans=-1;
        while(st<=en){
            int mid=st+(en-st)/2;
            if(nums[mid]==target){
                ans=mid;
                if(start==true){
                    en=mid-1;
                }
                else{
                    st=mid+1;
                }
            }
            else if(nums[mid]>target){
                en=mid-1;
            }
            else {
                st=mid+1;
            }
        }return ans;
    }
}

File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        vector <int> arr;
        
        // start
        int ans1=-1;
        int st=0,en=nums.size()-1;
        while(st<=en){
            int mid=st+(en-st)/2;
            if(nums[mid]==target){
                ans1=mid;
                en=mid-1;
            }
            else if(nums[mid]>target){
                en=mid-1;
                
            }
            else{
                st=mid+1;
            }
        }
        arr.push_back(ans1);
        if(ans1==-1){
            arr.push_back(-1);
            return arr;
        }
        // end
        // int ans2;
        ans1=0;
        st=0;
        en=nums.size()-1;
        while(st<=en){
            int mid=st+(en-st)/2;
            if(nums[mid]==target){
                ans1=mid;
                st=mid+1;
            }
            else if(nums[mid]>target){
                en=mid-1;
                
            }
            else{
                st=mid+1;
            }
        }
        arr.push_back(ans1);
        return arr;
        
    }
};